Review user guide (#979)
ansible/receptor: docs/source/user_guide/tls.rst

        port: 2222
        tls: myserver

> Having ``tls-server`` defined does nothing on its own, but it can be referenced elsewhere in the Receptor config file. Above, ``tls`` in the ``tcp-listener`` is set to use ``myserver``. In general, ``tls-server`` should be referenced anywhere Receptor is expecting an incoming connection, i.e. ``*-listener`` backends or on the ``control-service``. Similarly, ``tls-client`` should be referenced anywhere Receptor is expecting to make an outgoing connection, i.e. ``*-peer`` backends or in ``receptorctl`` (the command-line client for Receptor).

Reviewer: ```suggestion
Defining ``tls-server`` takes no effect, but can be referenced elsewhere in the Receptor config file.
In the preceding configuration snippet, ``tls`` in the ``tcp-listener`` is set to ``myserver``.
In general, ``tls-server`` should be referenced anywhere Receptor is expecting an incoming connection, such as ``*-listener`` backends or on the ``control-service``.
Similarly, ``tls-client`` should be referenced anywhere Receptor makes an outgoing connection, such as ``*-peer`` backends or in the ``receptorctl`` command-line client.
```

Generating certs
-----------------

> Receptor supports X.509 compliant certificates. Although numerous tools can be used to generate valid certificates, Receptor has a built-in tool to help with this process. Running Receptor with the ``cert-init``, ``cert-makereq``, and ``cert-signreq`` actions will create certificate authorities, make requests, and sign requests, respectively.

Reviewer: ```suggestion
Receptor supports X.509 compliant certificates and provides a built-in tool to generate valid certificates.
Running Receptor with the ``cert-init``, ``cert-makereq``, and ``cert-signreq`` actions creates certificate authorities, make requests, and sign requests.
```

      receptor --cert-signreq req=certs/$node.csr cacert=certs/ca.crt cakey=certs/ca.key outcert=certs/$node.crt
    done

> The above script will create a CA, and for each node `foo` and `bar`, create a certificate request and sign it with the CA. These certs and keys can then be used to create ``tls-server`` and ``tls-client`` definitions in the Receptor config files.

Reviewer: ```suggestion
The preceding script creates a CA, and for each node ``foo`` and ``bar``, create a certificate request and sign it with the CA.
These certificates and keys can then create ``tls-server`` and ``tls-client`` definitions in the Receptor config files.
```

Author: much nicer, thanks

Above the mesh TLS
-------------------

> Below-the-mesh TLS deals with connections that are being made to an IP address or DNS name, and so it can use normal X.509 certificates which include DNS names or IP addresses in their subjectAltName field.  However, above-the-mesh TLS deals with connections whose endpoint addresses are Receptor node IDs.  This requires generating certificates that include Receptor node IDs as names in the subjectAltName extension.  To do this, the otherName field of subjectAltName can be utilized.  This field is designed to accept arbitrary names of any type, and includes an ISO Object Identifier (OID) that defines what type of name this is, followed by arbitrary data that is meaningful for that type.  Red Hat has its own OID namespace, which is controlled by RHANANA, the Red Hat Assigned Names And Number Authority.  Receptor has an assignment within the overall Red Hat namespace.

Reviewer: ```suggestion
Below-the-mesh TLS deals with connections to an IP address or DNS name and can use normal X.509 certificates with DNS names or IP addresses in the ``subjectAltName`` field. 
Above-the-mesh TLS deals with connections that use Receptor node IDs as endpoint addresses, which requires generating certificates that include Receptor node IDs as names in the ``subjectAltName`` extension.
You can use the ``otherName`` field of ``subjectAltName`` to specify Receptor node IDs.
The ``otherName`` field accepts arbitrary names of any type, and includes an ISO Object Identifier (OID) that defines what type of name this is, followed by arbitrary data that is meaningful for that type.
Red Hat has its own OID namespace, which is controlled by RHANANA, the Red Hat Assigned Names And Number Authority.
Receptor has an assignment within the overall Red Hat namespace.
```

> If TLS authentication is used in your mesh, the certificates OIDs (1.3.6.1.4.1.2312.19.1) will be verified against the `node.id` specified in the configuration file. If there is no match, the Receptor binary will hard exit. To avoid this check, visit the `Skip Certificate Validation`_ section for more details.

Reviewer 2: Active voice:
```suggestion
If you use TLS authentication in your mesh, the certificates OIDs (1.3.6.1.4.1.2312.19.1) will be verified against the `node.id` specified in the configuration file. If there is no match, the Receptor binary will hard exit. To avoid this check, visit the `Skip Certificate Validation`_ section for more details.
```

Skip certificate validation
----------------------------

> Depending on the specifics of your environment(s), if certificate validation needs to be turned off, add a `skipreceptornamescheck` key-value pair in the configuration file for `tls-server`, `tls-config`, or both.

Reviewer 2: I'm not sure exactly what this means, would this be a correct rewriting?
```suggestion
You can turn off certificate validation by adding a `skipreceptornamescheck` key-value pair to your configuration. Depending on the specifics of your environment(s), you may need to add the `skipreceptornamescheck` key-value pair to the configuration file for `tls-server`, `tls-config`, or both.
```

Author: It's better but there may be some improvement possible in the future

> The default behaviour for this option is `false` which means that the certificate's OIDs will be verified against the node ID.

Reviewer 2: ```suggestion
The default behavior for this option is `false` which means that the certificate's OIDs will be verified against the node ID.
```
